### ISSUE TITLE
Add a way to include identity proofs in the blockchain

### ISSUE THREAD

**Issue by Insoleet** [duniter/sakia#616]
Following this post : https://forum.duniter.org/t/presentation-membre-tortue/1867/9?u=inso

This is a feature to consolidate the WoT with external proofs, like from centralized institutions like states.

I imagine the feature as the following : 
 - The user enters informations depending on his identity card in a dialog. For example, with a french id card, the dialog should be required to enter the following informations : 

```
Type: IDCARD
Version: 1
State: France
Numero: [Number of the idcard]
Nom: [Name]
Prenom: [Nickname]
Naissance: [Date of Birth]
Ville: [City]
Expiration: [Expiration date]
```

Note : The fields would probably change depending on the state issuing the IDCard. 

All the fields values should be uppercased with no special character. This text would then be hashed. A tx would be sent to himself, with the comment field fill with : 

```
IDTYPROOF:IDCARD:XHX(Hash)
```

Then, when an identity requires certification in real life, the other one can ask him to check for his idcard. The software compares the hash of the card shown by the identity with the hash present in the blockchain. 

 - If the hashes are the same, then for sure he is the same who published the identity
 - If the hash is different, there is a problem, and the identity should not be certified

The scenario is useful against sybil attacks. To try to get certifications on a second identity, the attacker needs another idcard. With the same idcard, the software will detect a hash colliding with his previous identity, and he would be discovered.

We should then provide multiple kind of documents to be able to provide multiple proofs when being certified by someone we do not know a lot. The documents required by a given community could change, but should be the same for everyone willing to join a given WoT.

**Comment by blavenie** [comment]
> Type: IDCARD
Version: 1
State: France
Numero: [Number of the idcard]
Nom: [Name]
Prenom: [Nickname]
Naissance: [Date of Birth]
Ville: [City]
Expiration: [Expiration date]

que penses tu de : 
```
Type: IDCARD
Version: 1
State: FR
Number: [Number of the idcard]
Lastname: [Name]
Firstname: [Nickname]
Birthday: YYYY-MM-DD
City: [City]
Expiration: [Expiration date]
```

> IDTYPROOF:IDCARD:XHX(Hash)

j'aurais ajouté le code ISO2 du pays + la version du doc. Par exemple : `IDTYPROOF:IDCARD:FR:1:XHX(Hash)`

**Comment by Insoleet** [comment]
> Pourquoi pas. Les noms en Français c'est pour correspondre exactement à la carte ?

Oui, ça me semblait plus logique. J'imagine que chaque carte d'identité possède des champs différents avec des termes qui peuvent être parfois techniques. La langue du document pour décrire les champs me parait cohérente du coup. 

> j'aurais ajouté le code ISO2 du pays + la version du doc. Par exemple : IDTYPROOF:IDCARD:FR:1:XHX(Hash)

Oui effectivement, ça semble logique pour les multinationaux.

**Comment by Tortue95** [comment]
Pour moi il ne faut pas que l'on utilise les champs: `"Number of the idcard"` et `"Expiration"` car il est possible de redemander une carte d'identité et permet donc d'ouvrir un nouveau compte membre avec un autre hash.
de plus cela permet de prouver son identité avec d'autre papiers tels que: permis, passeport, ...

j'ajouterais qu'il serait préférable que le champ `"nom"` soit le nom de jeune fille (ou jeune garçon) encore une fois pour éviter l'ouverture d'un autre compte membre après être marié.
ou d'avoir besoin de republier une transaction avec les nouvelles informations, ce qui risqueraient d'abaisser la confiance apportée.

je suis d'accord que ce n'est pas cool de devoir être obligé de passé par une reconnaissance de l’État.
mais je ne vois pas comment nous allons faire autrement pour le début de la vie de la monnaie.
dans les lieux où il y a moins de 5 membres proches les uns des autres.

**Comment by Insoleet** [comment]
Un probleme avec cette feature : quelqu'un qui se fait voler ses documents d'identité et dont le hash de ces docs se retrouve dans la blockchain apparaitra comme un sybil. Comment traiter cette situation ?

**Comment by Insoleet** [comment]
Après réflexion - ses nouveaux papiers d'identité posséderaient une date de révocation différente, ce qui règle le problème des collisions.

Autre point, cet outils est là pour permettre d'ajouter raisonablement de la confiance à la certification. Mais ça n'est pas suffisant pour des certifications de qualité sur Ğ1, si appliquées à un total inconnu. Ca peut aider à certifier sereinement quelqu'un de légèrement connu, disons.
